### PR TITLE
Add pagination support

### PR DESCRIPTION
**Pull request by codealchemy**
(Updated 2/5/14) This adds a Paginator class for paginating through call results.

Use:

```ruby
response = client.call(:people, :index)

paginated = Paginator.new(client, response)
page1 = paginated
page2 = page1.next
page3 = page2.next
```

The body of the response can be accessed through `.body`, in the above example, `page1.body` returns the same hash as `paginated`

Methods added:
- 'next' and 'prev' return the results of the next or previous page of results, nil if none
- 'next?' and 'prev?' return the path of the next page or nil depending on if there is another page available for the request.

@DavidHuie

### CODE REVIEW

File: lib/nationbuilder/client.rb [hunk 1]
```diff
@@ -68,6 +68,26 @@ def response
     Thread.current[:nationbuilder_rb_response]
   end
 
+  def next?
+    @next ? true : false
```
Comment (DavidHuie) [inline lib/nationbuilder/client.rb]: There's no need for the `? true : false` bit. If `@next` not set, that's considered a falsey value. The same applies to `prev?`

File: lib/nationbuilder/client.rb [hunk 2]
```diff
@@ -68,6 +68,26 @@ def response
...
+  end
+
+  def next
```
Comment (DavidHuie) [inline lib/nationbuilder/client.rb]: We should not add these methods to the client. I think a proper implementation of pagination would involve an external pagination class. The interface would be something similar to this:

```ruby
result = client.call(:basic_pages, :index, site_slug: 'organizeralexandreschmitt')
paginated = Paginator.new(result)
page1 = paginated
page2 = page1.next_page
page3 = page2.next_page
...
```

File: spec/nationbuilder_client_spec.rb [hunk 3]
```diff
@@ -3,8 +3,8 @@
 describe NationBuilder::Client do
 
   let(:client) do
-    NationBuilder::Client.new('dh',
-                              '6ed3ab9395daf0e3e39098761e85e8e703aa84921c5c1e32637f6984944cf1f2')
+    NationBuilder::Client.new('organizeralexandreschmitt',
+                              '53920a524356034a065515a37650df2bd295971975d5742b9daa50eb8c7404d5')
```
Comment (DavidHuie) [inline spec/nationbuilder_client_spec.rb]: Is this a working API key? You should deactivate it once you've generated the fixtures.

**Comment by DavidHuie** [comment]
This looks good. Thanks!

File: spec/nationbuilder_client_spec.rb (hunk 3 above)
Comment (codealchemy) [inline spec/nationbuilder_client_spec.rb]: Yeah, I revoked it after creating the fixtures

**Pull request merged**